### PR TITLE
Make docker defaults more useful for local testing setup and update documentation

### LINKED ISSUE #5008: TM not available after starting with docker-compose
I was trying to have first look at TM using docker / docker-compose. Specifically I used the manual at https://github.com/hotosm/tasking-manager/blob/develop/docs/setup-docker.md .

Basically I ran `docker-compose up -d`. I got the output

```
Starting postgresql                 ... done
Starting tasking-manager_frontend_1 ... done
Starting tasking-manager_traefik_1  ... done
Starting migration                  ... done
Starting backend                    ... done
```

but no service runs at http://127.0.0.1:5000 .

How can I debug this? The manual probably should also be amended with this information...

### PR DESCRIPTION
**Pull request by SlowMo24**
This PR updates the docker documentation that seems to lag behind the docker setup development. It does so by adapting the docker-compose defaults as well as the settings (.env) defaults.

In addition it updates and extends the documentation. This documentation update is quite brisk, so feel free to adapt anything where you don't agree. Especially I was not sure what would be the 'best-practice' to get administrator privileges on the self hosted Tasking Manager and how detailed to describe the procedure.

closes #5008

### CODE REVIEW

**Comment by SlowMo24** [comment]
sorry created this PR prematurely. Will rework and eventually reopen

**Comment by Aadesh-Baral** [comment]
Hi, @SlowMo24 we are working to implement OAuth2.0 for OSM login on #5029 which will have some changes to setup TM. Will review it as soon as #5029 is merged.

File: example.env [hunk 1]
```diff
@@ -5,15 +5,15 @@
 
 # The TM_APP_BASE_URL defines the URL of the frontend and is used by the backend
 # to configure links on emails and some authentication callbacks.
-# On development instances it should be 127.0.0.1:3000 or localhost:3000
+# On development instances it should be http://localhost:80
```
Comment (Aadesh-Baral) [inline example.env:8]:
```suggestion
# On development instances it should be 127.0.0.1:3000 or localhost:3000
# While setting up using `docker-compose` it should be http://localhost:80
```

File: example.env [hunk 2]
```diff
@@ -5,15 +5,15 @@
...
 # On production instances, use the public URL of your frontend
-TM_APP_BASE_URL=http://127.0.0.1:3000
+TM_APP_BASE_URL=http://localhost:80
```
Comment (Aadesh-Baral) [inline example.env:10]:
```suggestion
TM_APP_BASE_URL=http://127.0.0.1:3000
```

File: example.env [hunk 3]
```diff
@@ -5,15 +5,15 @@
...
 
 # The TM_APP_API_URL defines the URL of your backend server. It will be used by
 # both the backend and by the frontend
-# On development instances it should be 127.0.0.1:5000 or localhost:5000
+# On development instances it should be http://localhost:80
```
Comment (Aadesh-Baral) [inline example.env:14]:
```suggestion
# On development instances it should be 127.0.0.1:5000 or localhost:5000
# While setting up using `docker-compose` it should be http://localhost:80
```

File: example.env [hunk 4]
```diff
@@ -5,15 +5,15 @@
...
 # On production instances, use the public URL of your backend
-TM_APP_API_URL=http://127.0.0.1:5000
+TM_APP_API_URL=http://localhost:80
```
Comment (Aadesh-Baral) [inline example.env:16]:
```suggestion
TM_APP_API_URL=http://127.0.0.1:5000
```

**Review by Aadesh-Baral — CHANGES_REQUESTED** [review]
I think that adding frontend/backend URL instructions with and without docker to example.env would be useful. Other than that looks good to me.

**Comment by SlowMo24** [comment]
Thank you @Aadesh-Baral for reviewing. I have updated the config and docker-compose so it uses the same ports as the local development setup. I have also addressed #5313 and simplified the whole docker configuration. Feel free to comment.

**Review by Aadesh-Baral — APPROVED** [review]

**Comment by SlowMo24** [comment]
Hey, thanks for approving. I think there has been a rework of the documentation (there is now a `docs-old` folder in main) and there are merge-conflicts. Is there any action required from my side (i.e. should I have a look at the new docs?) or are you in the middle of a process there?

**Comment by Aadesh-Baral** [comment]
Hi @SlowMo24 , I've picked a commit from this PR to #5463 and planning to work on that PR itself.

**Comment by eternaltyro** [comment]
Related to #5705

**Review by eternaltyro — APPROVED** [review]
This looks good to be merged. cc : @Aadesh-Baral this seems to only affect docker-compose setups and might not interfere with other merge requests in queue. Can we proceed?

@SlowMo24 thank you very much for these changes. We will improve over them in due course, but this is a good starting point for us.

**Comment by eternaltyro** [comment]
Seems like there are some conflicts after-all. I will review this tomorrow and see how we can incorporate these changes into the new docs hierarchy.

**Comment by Aadesh-Baral** [comment]
> This looks good to be merged. cc : @Aadesh-Baral this seems to only affect docker-compose setups and might not interfere with other merge requests in queue. Can we proceed?

LGTM. Please go ahead.

**Comment by ramyaragupathy** [comment]
@mahesh-naxa - could you please verify if some of these issues have been fixed with the recent docker PRs? cc @spwoodcock @dakotabenjamin

**Comment by spwoodcock** [comment]
The actual config changes aren't required anymore - they have since been updated.

The docs in the PR were definitely useful at the time too, but I think the docs and config have changed since then, so it's hard to reconcile.

Unfortunately, it may be best to close this PR. The contribution is valuable - so sorry @SlowMo24 we didn't have the resource at the time to properly review and merge this 🙏

**Comment by mahesh-naxa** [comment]
His docs specifically on how to generate OSM Auth client/secret with screenshots are helpful imo. 
Others have been addressed.

**Comment by SlowMo24** [comment]
This MR was the result of me setting up my own tasking manager instance for a research project that I presented at SotM2022: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/ideal-vgi/osm-multitag . At that time I struggled quite a bit with the documentation but especially with Traefik (see #5313 ). The project is since completed and I currently don't work with it any more (:thinking: should take down that server I set up back then). So I am out of date with your code-base on the topic.

Things happen, let's move forward: I am happy to support you (I also gained a bit more knowledge on Docker and Traefik since then ;-) )

Please tell me where you could use support and I will see what I can do. For example I could run the setup procedure again with the current documentation and create issues or MRs where I fail. Yet, I would need some info on who you actually target with the documentation and the docker setup. If it is meant to be your team-documentation and should limit the [lottery factor](https://de.wikipedia.org/wiki/Truck_Number) we might think differently about it than if it was meant for what I did: random humans set up their own instance.
Or I could see If I can rebase this MR and reduce it the the parts that are still missing, if you think that is helpful.

**Comment by spwoodcock** [comment]
Thanks so much for the offer @SlowMo24, very kind of you 😄

Testing our current setup and providing feedback on the config or documentation around it would be a huge help 🙏

Feel free to rebase this or create a new MR/PR as needed & we will be much quicker to review next time!

**Pull request closed without merging**